crates/polars-core/src/fmt.rs

```diff
@@ -710,6 +724,15 @@ const SCIENTIFIC_BOUND: f64 = 999999.0;
 
 fn fmt_float<T: Num + NumCast>(f: &mut Formatter<'_>, width: usize, v: T) -> fmt::Result {
     let v: f64 = NumCast::from(v).unwrap();
+
+    let precision = get_float_precision();
+    if precision != u8::MAX {
```

Comment:
Instead of using`u8::MAX` to signal 'no option set' I think we should use an `Option<T>` initialized with `None`.

Reply:
@alicja-januszkiewicz: want to take this one? :)  I'm going to take a look at the max precision value again, as we should consider Decimal too (support for which is going to be turned on by default soon).

Reply:
Sure, I'll try to get this over by the end of the week.

Reply:
I've changed the type of `FLOAT_PRECISION` from `AtomicU8` to `RwLock<Option<usize>>`.

I've picked `usize` for now as that's the type the `write!` and `format!` macros expect, so this saves having to repeatedly cast the value.

I've wrapped it in an `RwLock` to hopefully retain whatever protection `AtomicU8` was providing, as there doesn't seem to be an atomic equivalent of `Option`.